core/src/labels/labelCollider.cpp

```diff
@@ -251,6 +265,8 @@ void LabelCollider::process(TileID _tileID, float _tileInverseScale, float _tile
         }
     }
 
+    filterRepeatGroups(repeatPos, m_labels.size()-1);
```

Comment:
The idea is to filter out repeated labels early so that they don't block lower priority labels.

hold on - updating the PR to address some comments